R/demo-crowding.R

```diff
@@ -67,15 +69,29 @@ demo_crowding <- function(area, areatype, crowding, race) {
       output = 1
     ) %>%
     req_perform()
-
-  resp <- process_resp(resp, "demographics")
-
-  resp %>%
-    setNames(c(
-      get_area(areatype),
-      "Percent",
-      "Households",
-      "Rank"
-    )) %>%
-    mutate(across(c("Percent", "Households"), \(x) as.numeric(x)))
+  
+  resp <- process_resp(resp, "demographics", include_metadata)
+  
+  if (include_metadata == TRUE) {
+    resp$data <- resp$data %>%
+      setNames(c(
+        get_area(areatype),
+        "Percent",
+        "Households",
+        "Rank"
+      )) %>%
+      mutate(across(c("Percent", "Households"), \(x) as.numeric(x)))
+    
+    result_df <- process_metadata(resp)
+    
+  } else {
+    resp %>%
+      setNames(c(
+        get_area(areatype),
+        "Percent",
+        "Households",
+        "Rank"
+      )) %>%
+      mutate(across(c("Percent", "Households"), \(x) as.numeric(x)))
```

Comment:
It isn't immediately clear that this function returns `result_df` if `include_metadata == TRUE` and returns the output of the else statement if `include_metadata == FALSE`.

I would assign the output of running the `dplyr` verbs to `result_df` and then return `result_df` at the very last line of this function.